genie-proto/src/main/proto/genie.proto

```diff
@@ -264,3 +265,51 @@ message JobKillRegistrationRequest {
 
 message JobKillRegistrationResponse {
 }
+
+//--------------------------------------------------------------------
+// Agent Job File Service
+//
+// This service streams files from an agent running a job to the server
+// so they can be served via API.
+//--------------------------------------------------------------------
+
+service JobFileService {
+    rpc pushManifest (stream JobFileManifestMessage) returns (stream google.protobuf.Empty);
+    rpc streamFile (stream AgentJobFileMessage) returns (stream ServerJobFileMessage);
+}
+
+message JobFileManifestMessage {
+    string job_id = 1;
+    string manifest_json_string = 2;
```

Comment:
Are you suggesting making the field type `bytes` and then using java built-in object serialization to serialize the manifest object?

Reply:
haha oh god no. Just saying there's a `ObjectMapper.writeValueAsBytes()` and you could use a gRPC `ByteString`. You could also leave it as string and still enable compression.

Reply:
Out of curiosity, i tried to serialize a manifest as `String`, vs `bytes` to see how much of a difference it would make...

```
String size: 10565
Bytes size: 10565
```

While at it, i did a quick micro-benchmark to see if one is faster than the other...

```
Serialized 1000 times to string in 238ms
Serialized 1000 times to bytes in 212ms
```

So given these numbers I'd rather leave it as-is, since it's human readable at no cost in size or time.